### ISSUE TITLE
Sequence Diagram - conversation with more than 50 messages

### ISSUE THREAD

**Issue by mikeminutillo** [Particular/ServiceInsight#510]
The Sequence Diagram only shows 50 messages. If there are more than this, some of them are not shown. 

This can cause an issue when one of the messages not shown in the diagram is the one selected in the grid at the top.

<img width="578" alt="morethan50messagesinconversation" src="https://cloud.githubusercontent.com/assets/124014/11328693/3514715c-91ca-11e5-9d58-ad35b4e5917e.PNG">

## Repro
* Create a conversation with more than 50 messages.
* Select a message in ServiceInsight which is a part of the conversation
* Select each message in the list view until one is found which is not displayed on the diagram
* What happens? No message is selected in the diagram
* What should happen? The message which has been selected should appear on the diagram

We also don't show any indication to the user that part of the conversation is missing.

**Comment by distantcam** [comment]
I suspect this is a SC issue. Do you know if the SC endpoint is returning all the messages in the conversation, or just the 50? If it's paging is there a way to get the other pages?

**Comment by mikeminutillo** [comment]
@distantcam Yeah, SC is paging the results. You can specify a number of parameters to increase the page size, select which page you want, and which order you want. It will presumably need a UI component as well to show the user that not all messages are displayed and the next page is needed.

**Comment by distantcam** [comment]
@mikeminutillo We already show paged results for the search, but I think for diagrams we should just automatically grab all the pages when showing the diagram.

**Comment by HEskandari** [comment]
@distantcam not practically possible in all cases...e.g. never ending sagas, to name one.

**Comment by johnsimons** [comment]
> but I think for diagrams we should just automatically grab all the pages when showing the diagram.

You can't do that @distantcam, there are very long (multiple years of messages) conversations

**Comment by johnsimons** [comment]
We just need to have a paging way for our diagrams

**Comment by distantcam** [comment]
> We just need to have a paging way for our diagrams

That's fine for some diagrams, but might be problematic for other diagrams. I guess we can cross that bridge when we get to it.

**Comment by johnsimons** [comment]
Agree @distantcam

**Comment by mikeminutillo** [comment]
If we were explicit about capturing the end of the conversation (i.e. the most recently processed messages) then we could just show a big button down the left hand side that said "more" and continually load the next page going back in time. That has the issue that a new message coming in will throw off the paging but that just means we might end up with 99 msgs instead of 100 and I doubt the user would know/care. 

Would you rearrange the whole diagram when you load more data? i.e. would the endpoints shuffle around?

**Comment by tmasternak** [comment]
Removing `Tag: Consider for Short Term Focus`. @HEskandari  Please add `Consider for Promotion` if it's still relevant.

**Comment by kbaley** [comment]
We have triaged this issue and will be considering it for inclusion in one of the upcoming maintenance releases for this package.

**Comment by HEskandari** [comment]
The default is changed to 200 now and can be overridden from an `app.config` value. The underlying issue of not loading paged data is still there.
